integration-libs/cdp/root/service/cdp-order-service.ts

```diff
@@ -4,19 +4,191 @@
+  }
+
+  public fetchOrder(page: number, page_size: number): Observable<finalOrder> {
+    this.obser$ = this.userIdService
+      .takeUserId()
+      .pipe(
+        switchMap((userId) =>
+          this.cdpOrderAdapter.getOrderPerPage(userId, page_size, page)
+        )
+      );
+    return this.obser$;
+  }
+
+  public fetchReturn(): Record<string, string>{
+    this.returnObser$=this.userIdService.takeUserId().pipe(switchMap((userId) => this.cdpOrderAdapter.getRetunDetail(userId)));
+    //this.returnObser$=this.cdpOrderAdapter.getRetunDetail();
+    this.returnObser$.subscribe((returnSub)=>{
+      if(returnSub.returnRequest!=null){
+        returnSub.returnRequest.forEach((returnval)=>{
+          this.returnDate[returnval.ReefernceSDDocumentID]=returnval.creationTime;
+      });
+      }
+    });
+    return this.returnDate;
+  }
+
+  public async fetchOrderDetail(
+    finalResult: finalOrder
+  ): Promise<Record<string, order>> {
+    for (let orderval of finalResult.orders) {
+      await this.userIdService
+        .takeUserId()
+        .pipe(
+          mergeMap((userId) =>
+            this.cdpOrderAdapter.getOrderDetail(userId, orderval)
+          )
+        )
+        .toPromise()
+        .then((data) => {
+          this.orderDetail[orderval.code] = data;
+        });
+    }
+    return this.orderDetail;
+  }
+
+  public fetchOrderStatus(
+    detail: Record<string, order>
+  ): Record<string, Record<string, number>> {
+    this.clearCart();
+    // eslint-disable-next-line guard-for-in
+    for (let orderCode in detail) {
+      this.orderStatus[orderCode] ??= {};
+      if(detail[orderCode].consignments!=null)
+      {
+        detail[orderCode].consignments.forEach((ord) => {
+          this.orderStatus[orderCode][ord.status] ??= 0;
+          ord.entries.forEach((entr) => {
+            this.orderStatus[orderCode][ord.status] =
+              this.orderStatus[orderCode][ord.status] + entr.quantity;
+          });
+        });
+      }else if(detail[orderCode].deliveryStatus!=null)
+      {
+        this.orderStatus[orderCode][detail[orderCode].deliveryStatus] ??= 0;
+        detail[orderCode].entries.forEach((entr) => {
+          this.orderStatus[orderCode][detail[orderCode].deliveryStatus] =
+            this.orderStatus[orderCode][detail[orderCode].deliveryStatus] + entr.quantity;
+        });
+      }
+
+      //setting the order status for Returned
+      if (this.returnDate.hasOwnProperty(orderCode)) {
+        const value = this.returnDate[orderCode];
+        if (value.trim() === '') {
+          console.log('The value for someKey is empty.');
+        } else {
+          this.orderStatus[orderCode]["RETURNED"] ??= 0;
+          this.orderStatus[orderCode]["RETURNED"]=1;
+        }
+        } else {
+          console.log('There is no property with the key someKey in the returnDate object.');
+      }
+    }
+    console.log(this.orderStatus);
+    return this.orderStatus;
+  }
+
+  public fetchOrderImage(
```

Comment:
This method can be modified to be invoked immediately for each order-detail. In this way, we can avoid the Object

We can discuss.